pkg/api/grpc/daprinternal.go

```diff
@@ -186,12 +188,46 @@ func (a *api) CallLocalStream(stream internalv1pb.ServiceInvocation_CallLocalStr
 		pw.Close()
 	}()
 
+	header := http.Header{}
+	for k, v := range chunk.GetRequest().GetMetadata() {
+		for _, vv := range v.GetValues() {
+			header.Add(k, vv)
+		}
+	}
+
+	isSSE := false
+	// Or check the Accept header
+	if accept := header.Get("Accept"); strings.Contains(accept, "text/event-stream") {
+		isSSE = true
+		respwritter := loggingResponseWriter{
+			header: header,
+			stream: stream,
+			appID:  a.AppID(),
+		}
+		req.WithHTTPResponseWriter(&respwritter)
+	}
+
 	// Submit the request to the app
 	res, err := appChannel.InvokeMethod(ctx, req, "")
 	if err != nil {
 		statusCode = int32(codes.Internal)
 		return status.Errorf(codes.Internal, messages.ErrChannelInvoke, err)
 	}
+
+	if res == nil && isSSE {
+		return nil
+	}
+
+	if res != nil && isSSE {
+		statusOK := res.Status().GetCode() >= 200 && res.Status().GetCode() < 300
+		msg := "no response received from stream"
+		if statusOK {
+			msg = "no expected response from stream"
+		}
+		statusCode = int32(codes.Internal)
+		return status.Errorf(codes.Internal, messages.ErrChannelInvoke, errors.New(msg))
+	}
```

Comment:
I think we need to call `res.Close` in this case. Maybe realigning the ifs so the `defer res.Close()` is called?
Something like:
```go
if res == nil
	if isSSE {
		return nil
	}
	statusCode = int32(codes.Internal)
	return status.Errorf(codes.Internal, messages.ErrChannelInvoke, errors.New("no response received from stream"))
}

defer res.Close()

if isSSE {
	statusOK := res.Status().GetCode() >= 200 && res.Status().GetCode() < 300
	msg := "no response received from stream"
	if statusOK {
		msg = "no expected response from stream"
	}
	statusCode = int32(codes.Internal)
	return status.Errorf(codes.Internal, messages.ErrChannelInvoke, errors.New(msg))
}
```